modules/swagger-parser-v3/src/main/java/io/swagger/v3/parser/util/OpenAPIDeserializer.java

```diff
@@ -1408,13 +1412,25 @@ else if(!v.isValueNode()) {
         }
         Set<String> filter = new HashSet<>();
 
-        for(Parameter param:parameters) {
+        parameters.stream().map(this::getParameterDefinition).forEach(param -> {
             if(!filter.add(param.getName()+"#"+param.getIn())) {
```

Comment:
@gracekarina , this is not working when the parameter is coming from another file. param.getName() will always return null in this case, so it won't pass the next filter.

Reply:
Hi please, send us a spec where this will fail so we can test and addressed the issue.

Reply:
File one:
```
openapi: 3.0.0
info:
  version: 1.0.0
  title: an API
  description: An API for reproduce a *There are duplicate parameter values*.

paths:
  /anPath:
    get:
      parameters:
        - $ref: './references.oas3.yaml#/components/parameters/ParamQ_One'
        - $ref: './references.oas3.yaml#/components/parameters/ParamQ_Two'
      responses:
        '200':
          description: OK
          content:
            application/json:
              schema:
                type: array
                items:
                  type: string
```
file references.oas3.yaml:
```
components:
  parameters:
    ParamQ_One:
      in: query
      name: customer-id
      required: true
      schema:
        format: int32
        type: integer
    ParamQ_Two:
      in: query
      name: unit-id
      schema:
        format: int32
        type: integer
```

Reply:
well, to resolve parameters it should be done after deserializing in the resolver part. I'm looking at it, but this is a different issue.

Reply:
@gracekarina, 
well, could be, but I don't think it is a different issue...
we'll still get 'duplicated values' because a parameter is only unique when we have its name + its type. if we don't have its type then the problem of duplicate will arise and will preventing any generator to work.

Reply:
What could be done is validate in deserializing time, if the ref in the parameter is internal, or remote so we would avoid the message warning.

Reply:
Agreed, and I don't see any option other than to resolve that $ref at this deserializing time... :)

Reply:
merged PR https://github.com/swagger-api/swagger-parser/pull/1080

Reply:
@gracekarina, well done, girl !